File: core/src/test/java/com/pipai/adv/utils/DirectionUtilsTest.java
package com.pipai.adv.utils;

import org.junit.Assert;
import org.junit.Test;

import com.pipai.adv.backend.battle.domain.Direction;

public class DirectionUtilsTest {

    @Test
    public void testDirectionFor() {
        Assert.assertEquals(Direction.N, DirectionUtils.INSTANCE.directionFor(0, 0, 0, 1));
        Assert.assertEquals(Direction.S, DirectionUtils.INSTANCE.directionFor(0, 0, 0, -1));
        Assert.assertEquals(Direction.E, DirectionUtils.INSTANCE.directionFor(0, 0, 1, 0));
        Assert.assertEquals(Direction.W, DirectionUtils.INSTANCE.directionFor(0, 0, -1, 0));
        Assert.assertEquals(Direction.NE, DirectionUtils.INSTANCE.directionFor(0, 0, 1, 1));
        Assert.assertEquals(Direction.NW, DirectionUtils.INSTANCE.directionFor(0, 0, -1, 1));
        Assert.assertEquals(Direction.SE, DirectionUtils.INSTANCE.directionFor(0, 0, 1, -1));
        Assert.assertEquals(Direction.SW, DirectionUtils.INSTANCE.directionFor(0, 0, -1, -1));
    }

}

File: core/src/test/java/com/pipai/adv/utils/DirectionUtilsTest.kt
package com.pipai.adv.utils

import org.junit.Assert
import org.junit.Test

import com.pipai.adv.backend.battle.domain.Direction

class DirectionUtilsTest {

    @Test
    fun testDirectionFor() {
        Assert.assertEquals(Direction.N, DirectionUtils.directionFor(0f, 0f, 0f, 1f))
        Assert.assertEquals(Direction.S, DirectionUtils.directionFor(0f, 0f, 0f, -1f))
        Assert.assertEquals(Direction.E, DirectionUtils.directionFor(0f, 0f, 1f, 0f))
        Assert.assertEquals(Direction.W, DirectionUtils.directionFor(0f, 0f, -1f, 0f))
        Assert.assertEquals(Direction.NE, DirectionUtils.directionFor(0f, 0f, 1f, 1f))
        Assert.assertEquals(Direction.NW, DirectionUtils.directionFor(0f, 0f, -1f, 1f))
        Assert.assertEquals(Direction.SE, DirectionUtils.directionFor(0f, 0f, 1f, -1f))
        Assert.assertEquals(Direction.SW, DirectionUtils.directionFor(0f, 0f, -1f, -1f))
    }

}
